lib/src/constants/revive.dart

```diff
@@ -22,10 +22,26 @@ Revivable reviveInstance(DartObject object, [LibraryElement origin]) {
   origin ??= object.type.element.library;
   var url = Uri.parse(urlOfElement(object.type.element));
   final clazz = object?.type?.element as ClassElement;
-  for (final e in clazz.fields.where(
-    (f) => f.isPublic && f.isConst && f.computeConstantValue() == object,
-  )) {
-    return new Revivable._(source: url, accessor: e.name);
+  // Enums are not included in .definingCompilationUnit.types.
+  if (clazz.isEnum) {
+    for (final e in clazz.fields.where(
+        (f) => f.isPublic && f.isConst && f.computeConstantValue() == object)) {
+      return new Revivable._(
+        source: url.removeFragment(),
+        accessor: '${clazz.name}.${e.name}',
+      );
+    }
+  }
+  for (final e in origin.definingCompilationUnit.types
+      .map((c) => c.fields)
+      .expand((f) => f)
```

Comment:
[nit] `origin.definingCompilationUnit.types.expand((c) => c.fields)` rather than a separate map and expand step

Reply:
Done.